File: src/org/usfirst/frc5293/commands/util/ActionCommandGroup.java
package org.usfirst.frc5293.commands.util;

import edu.wpi.first.wpilibj.command.Subsystem;

import java.util.List;

public abstract class ActionCommandGroup extends EmptyCommandGroup {

    protected boolean isFinished = false;

    protected ActionCommandGroup() {
    }

    protected ActionCommandGroup(Subsystem subsystem) {
        requires(subsystem);
    }

    protected ActionCommandGroup(List<Subsystem> subsystems) {
        for (Subsystem subsystem : subsystems) {
            requires(subsystem);
        }
    }

    @Override
    protected boolean isFinished() {
        return isFinished;
    }
}

File: src/org/usfirst/frc5293/commands/util/ActionCommandGroup.kt
package org.usfirst.frc5293.commands.util

import edu.wpi.first.wpilibj.command.Subsystem

abstract class ActionCommandGroup : EmptyCommandGroup {

    private var _isFinished = false

    protected constructor(vararg subsystems: Subsystem) {
        subsystems.forEach { requires(it) }
    }

    override fun isFinished(): Boolean {
        return _isFinished
    }

    protected fun done() {
        _isFinished = true
    }
}
